include/picongpu/fields/background/templates/twtsfast/BField.hpp

```diff
@@ -0,0 +1,182 @@
+/* Copyright 2014-2020 Alexander Debus, Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include <pmacc/types.hpp>
+
+#include <pmacc/math/Vector.hpp>
+#include <pmacc/dimensions/DataSpace.hpp>
+#include "picongpu/fields/background/templates/twtsfast/numComponents.hpp"
+
+namespace picongpu
+{
+    /* Load pre-defined background field */
+    namespace templates
+    {
+        /* Traveling-wave Thomson scattering laser pulse */
+        namespace twtsfast
+        {
+            class BField
+            {
+            public:
+                using float_T = float_X;
+
+                enum PolarizationType
+                {
+                    /* The linear polarization of the TWTS laser is defined
```

Comment:
Doxygen comments should start with `/**` 
please update all otjer commemts too.

Reply:
Done.